shell.nix

```
@@ -0,0 +1,36 @@
+{ pkgs ? import <nixpkgs> {} } :
+let
+	langShell = lang: (import (./. + ("/" + lang + "/shell.nix")) { inherit pkgs; });
+	# We need this mess of no-op concatenations because Debian 11/Bullseye Stable is stuck on Nix 2.3.7 from 2020, which apparently doesn't always handle literals and antiquotation well:
+	# https://gist.github.com/CMCDragonkai/de84aece83f8521d087416fa21e34df4/fde9346664741a18d2748a578d9a1b648ee42dbd
+	combineShells = shells: pkgs.mkShell (builtins.foldl'
+		(shellA: shellB: 
+			{
+				buildInputs = shellA.buildInputs ++ shellB.buildInputs;
+			}
+		)
+		(pkgs.mkShell { inherit pkgs; })
+		shells
+	);
```

I think this is a good use case for [`inputsFrom`](https://github.com/NixOS/nixpkgs/blob/0bda862edc8b106779faafa4b5467f5fb85f3362/pkgs/build-support/mkshell/default.nix#L8-L9) on `mkShell`; i.e. I think you can replace the above with:

```suggestion
	combineShells = shells: pkgs.mkShell { inputsFrom = shells; };
```